### PR TITLE
pageserver: always detach before deleting

### PR DESCRIPTION
**Pull request by jcsp**
In #7957 we enabled deletion without attachment, but retained the old-style deletion (return 202, delete in background) for attached tenants.  In this PR, we remove the old-style deletion path, such that if the tenant delete API is invoked while a tenant is detached, it is simply detached before completing the deletion.

This intentionally doesn't rip out all the old deletion code: in case a deletion was in progress at time of upgrade, we keep around the code for finishing it for one release cycle.  The rest of the code removal happens in https://github.com/neondatabase/neon/pull/8091

Now that deletion will always be via the new path, the new path is also updated to use some retries around remote storage operations, to tripping up the control plane with 500s if S3 has an intermittent issue.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pageserver/src/http/routes.rs [hunk 1]
```diff
@@ -1071,23 +1068,16 @@ async fn tenant_delete_handler(
...
-        .delete_tenant(tenant_shard_id, ACTIVE_TENANT_TIMEOUT)
+        .delete_tenant(tenant_shard_id)
         .instrument(info_span!("tenant_delete_handler",
             tenant_id = %tenant_shard_id.tenant_id,
             shard_id = %tenant_shard_id.shard_slug()
         ))
         .await?;
 
-    // Callers use 404 as success for deletions, for historical reasons.
-    if status == StatusCode::NOT_FOUND {
```
Comment (arpad-m) [inline pageserver/src/http/routes.rs:1084]: if you remove this you should change the spec yml file, which currently says 404 is a success result. I mean it still kind of is but not as much as it used to any more.

Comment (jcsp) [reply pageserver/src/http/routes.rs:1084]: True - f27d24af

**Comment by jcsp** [comment]
(checking in with control plane folks that we're safe to merge this https://neondb.slack.com/archives/C03438W3FLZ/p1718966249028299?thread_ts=1718618574.225409&cid=C03438W3FLZ)

**Review by arpad-m — APPROVED** [review]

**Review by problame — APPROVED** [review]
Nice cleanup!

**Pull request merged**